examples/defaulter-gen/generators/defaulter.go

```diff
@@ -271,8 +281,33 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 			return false
 		}
 
+		// if the types are not in the same package where the defaulter functions to be generated
+		inputTags := extractInputTag(pkg.Comments)
+		if len(inputTags) > 1 {
+			panic(fmt.Sprintf("there could only be one input tag, got %#v", inputTags))
+		}
+		if len(inputTags) == 1 {
+			inputDir := filepath.Join(pkg.Path, inputTags[0])
+			err := context.AddDir(inputDir)
+			if err != nil {
+				glog.Fatalf("cannot import package %s", inputTags[0])
+			}
+			inputPkgPath, err := context.DirectoryToPackagePath(inputDir)
+			if err != nil {
+				glog.Fatalf("cannot translate dir %s to package path package %s", inputDir)
+			}
+			var ok bool
+			typesPkg, ok = context.Universe[inputPkgPath]
```

Comment:
context.Universe[inputPkgPath] returns k8s.io/gengo/types#Package.
`context.AddDirTo` calls builder to parse the source code, which only knows go/build#Package and go/types#Package.

The translation from directory to canonical path is done in builder, i can add a method to it to export the functionality, and then export this method via context. But I think that is worse than just return the package path via `AddDirTo`.